pkg/scheduler/eventhandlers_test.go

```diff
@@ -42,18 +45,173 @@ import (
+			updateFunc: func(s *Scheduler) {
+				s.deletePodFromSchedulingQueue(medPriorityPod)
+			},
+			wantInActive: []string{lowPriorityPod.Name},
+		},
+		{
+			name: "Addition of a nominated node name to the high priority pod that did not have it before shouldn't trigger rescheduling",
+			updateFunc: func(s *Scheduler) {
+				updatedPod := highPriorityPod.DeepCopy()
+				updatedPod.Status.NominatedNodeName = "node2"
+				updatedPod.ResourceVersion = "1"
+				s.updatePodInSchedulingQueue(highPriorityPod, updatedPod)
+			},
+			wantInActive: []string{},
+		},
+	}
+
+	for _, tt := range tests {
+		for _, qHintEnabled := range []bool{false, true} {
+			t.Run(fmt.Sprintf("%s, with queuehint(%v)", tt.name, qHintEnabled), func(t *testing.T) {
+				featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.SchedulerQueueingHints, qHintEnabled)
+
+				logger, ctx := ktesting.NewTestContext(t)
+				ctx, cancel := context.WithCancel(ctx)
+				defer cancel()
+
+				var objs []runtime.Object
+				for _, pod := range unschedulablePods {
+					objs = append(objs, pod)
+				}
+				client := fake.NewClientset(objs...)
+				informerFactory := informers.NewSharedInformerFactory(client, 0)
+
+				recorder := metrics.NewMetricsAsyncRecorder(3, 20*time.Microsecond, ctx.Done())
+				queue := internalqueue.NewPriorityQueue(
+					newDefaultQueueSort(),
+					informerFactory,
+					internalqueue.WithMetricsRecorder(*recorder),
+					internalqueue.WithQueueingHintMapPerProfile(queueingHintMap),
+					// disable backoff queue
+					internalqueue.WithPodInitialBackoffDuration(0),
+					internalqueue.WithPodMaxBackoffDuration(0))
+				schedulerCache := internalcache.New(ctx, 30*time.Second)
+
+				// Put test pods into unschedulable queue
+				for _, pod := range unschedulablePods {
+					queue.Add(logger, pod)
+					poppedPod, err := queue.Pop(logger)
+					if err != nil {
+						t.Fatalf("Pop failed: %v", err)
+					}
+					poppedPod.UnschedulablePlugins = sets.New("fooPlugin1")
+					queue.AddUnschedulableIfNotPresent(logger, poppedPod, queue.SchedulingCycle())
+				}
+
+				s, _, err := initScheduler(ctx, schedulerCache, queue, client, informerFactory)
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				if len(s.SchedulingQueue.PodsInActiveQ()) > 0 {
+					t.Error("No pods were expected to be in to activeQ before the update, but there were some", "actual", s.SchedulingQueue.PodsInActiveQ())
+				}
+				tt.updateFunc(s)
+				if len(s.SchedulingQueue.PodsInActiveQ()) != len(tt.wantInActive) {
+					t.Error("Different number of pods were expected to be in to activeQ, but it wasn't", "actual", s.SchedulingQueue.PodsInActiveQ(), "expected", tt.wantInActive)
+				}
+				for _, pod := range queue.PodsInActiveQ() {
+					if !isNameOnTheList(pod.Name, tt.wantInActive) {
+						t.Error("Found unexpected pod in activeQ", "pod", pod.Name)
+					}
+				}
+			})
+		}
+	}
+}
+
+func isNameOnTheList(str string, list []string) bool {
```

Comment:
Should we use the set utils from https://github.com/kubernetes/kubernetes/blob/25be4a4bd409249ce9f1289b3cd16b3119df80f4/staging/src/k8s.io/apimachinery/pkg/util/sets/set_generic_test.go#L27-L37